### PR TITLE
Terminals getters added to Bus interface

### PR DESCRIPTION
**Pull request by floriand-e2r**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
Connected terminals getters exist but in the BusExt interface (impl package)

**What is the new behavior (if this is a feature change)?**
The two getters are added in the Bus interface. 

**Does this PR introduce a breaking change or deprecate an API?** No
*If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:
These getters are needed for using the SlackTerminal introduced by issue #1307

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java [hunk 1]
```diff
@@ -240,4 +240,14 @@
      */
     void visitConnectedOrConnectableEquipments(TopologyVisitor visitor);
 
+    /**
+     * Get an iterable of the terminals connected to the bus.
+     */
+    Iterable<? extends Terminal> getConnectedTerminals();
```
Comment (mathbagu) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java:246]: I think you should to provide a default behavior to not break existing code (powsybl-network-store)

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java [hunk 2]
```diff
@@ -240,4 +240,14 @@
...
+
+    /**
+     * Get a stream of the terminals connected to the bus.
+     */
+    Stream<? extends Terminal> getConnectedTerminalStream();
```
Comment (mathbagu) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java:251]: Same remark

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java [hunk 3]
```diff
@@ -240,4 +241,18 @@
      */
     void visitConnectedOrConnectableEquipments(TopologyVisitor visitor);
 
+    /**
+     * Get an iterable of the terminals connected to the bus.
+     */
+    default Iterable<? extends Terminal> getConnectedTerminals() {
+        return new LinkedList<>();
```
Comment (geofjamg) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java:248]: Why a LinkedList and not an ArrayList?

Comment (floriand-e2r) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java:248]: An old pre-Java 8 habit, as it is a constant empty list. Thus it does not require the whole ArrayList mechanism.

**Comment by floriand-e2r** [comment]
@geofjamg what do you think about the code smell "Generic wildcard types should not be used in return types"? Here that allowed me to have a minimal influence on the Bus-inheriting classes, which can then have a list of TerminalExt and use TerminalExt methods. They don't though: all the calls to getConnectedTerminals() or getConnectedTerminalStream() can easily deal with Terminal instead of TerminalExt.

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java (hunk 3 above)
Comment (floriand-e2r) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Bus.java]: replaced by `Collections.emptyList()` (and not the raw type `Collections.EMPTY_LIST`)

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusExt.java [hunk 4]
```diff
@@ -16,8 +16,10 @@
  */
 interface BusExt extends Bus {
 
+    @Override
```
Comment (geofjamg) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusExt.java:19]: To remove as it is already in Bus?

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractBus.java [hunk 5]
```diff
@@ -52,8 +52,10 @@ public VoltageLevel getVoltageLevel() {
         return voltageLevel;
     }
 
+    @Override
```
Comment (geofjamg) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractBus.java:55]: Not necessary as it is already added to ConfiguredBusImpl and CalculaedBusImpl

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusExt.java (hunk 4 above)
Comment (floriand-e2r) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusExt.java:19]: Leaving it forced the classes inheriting BusExt to use TerminalExt, and not another class extending Terminal.

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractBus.java (hunk 5 above)
Comment (floriand-e2r) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractBus.java:55]: It is already added in ConfiguredBusImpl and CalculatedBusImpl but the methods using it in AbstractBus assume it returns a TerminalExt and not a Terminal. As mentionned earlier, these methods can deal with a Terminal though.

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusExt.java (hunk 4 above)
Comment (floriand-e2r) [reply iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BusExt.java]: For instance, with latest commit removing it, MergedBus.getConnectedTerminals() is now returning Iterable<Terminal> and not Iterable<TerminalExt>, and dealing with ConfiguredBus.getConnectedTerminals() hence with terminals whose type is `? extends Terminal`.

**Pull request merged**